Lib/test/test_xml_etree.py

```diff
@@ -692,28 +692,33 @@ def test_iterparse(self):
         it = iterparse(TESTFN)
         action, elem = next(it)
         self.assertEqual((action, elem.tag), ('end', 'document'))
-        with warnings_helper.check_no_resource_warning(self):
-            with self.assertRaises(ET.ParseError) as cm:
-                next(it)
-            self.assertEqual(str(cm.exception),
-                    'junk after document element: line 1, column 12')
+        with self.assertRaises(ET.ParseError) as cm:
+            next(it)
+        self.assertEqual(str(cm.exception),
+                'junk after document element: line 1, column 12')
+        with self.assertWarns(ResourceWarning):
             del cm, it
+            gc_collect()
 
-        # Not exhausting the iterator still closes the resource (bpo-43292)
-        with warnings_helper.check_no_resource_warning(self):
+        # Deleting iterator without close() should emit ResourceWarning (bpo-43292)
```

Comment:
Restore the old comment. The purpose of this test is to check that the resource is closed.